src/main/fc/rc.c

```diff
@@ -496,134 +496,182 @@ NOINLINE void initAveraging(uint16_t feedforwardAveraging)
     static bool prevDuplicatePacket[3];             // to identify multiple identical packets
-    static uint16_t feedforwardAveraging = 0;
-
+    static uint16_t feedforwardAveraging;
     if (feedforwardAveraging != pid->feedforwardAveraging) {
         feedforwardAveraging = pid->feedforwardAveraging;
         initAveraging(feedforwardAveraging);
     }
 
-    const float rcCommandDeltaAbs = fabsf(rcCommand[axis] - prevRcCommand[axis]);
+    const float rcCommandDelta = rcCommand[axis] - prevRcCommand[axis];
     prevRcCommand[axis] = rcCommand[axis];
+    float rcCommandDeltaAbs = fabsf(rcCommandDelta);
 
-    float setpoint = rawSetpoint[axis];
-    float setpointSpeed = (setpoint - prevSetpoint[axis]);
+    const float setpoint = rawSetpoint[axis];
+    const float setpointDelta = (setpoint - prevSetpoint[axis]);
     prevSetpoint[axis] = setpoint;
-    float setpointAcceleration = 0.0f;
-
-
-    if (axis == FD_ROLL) {
-        DEBUG_SET(DEBUG_FEEDFORWARD, 0, lrintf(setpoint * 10.0f)); // un-smoothed final feedforward
-    }
-
-    // attenuators
-    float zeroTheAcceleration = 1.0f;
-    float jitterAttenuator = 1.0f;
-    if (pid->feedforwardJitterFactor) {
-        if (rcCommandDeltaAbs < pid->feedforwardJitterFactor) {
-            jitterAttenuator = MAX(1.0f - (rcCommandDeltaAbs + prevRcCommandDeltaAbs[axis]) * pid->feedforwardJitterFactorInv, 0.0f);
-            // note that feedforwardJitterFactorInv includes a divide by 2 to average the two previous rcCommandDeltaAbs values
-            jitterAttenuator = 1.0f - jitterAttenuator * jitterAttenuator;
-        }
-    }
-    prevRcCommandDeltaAbs[axis] = rcCommandDeltaAbs;
 
-    // interpolate setpoint if necessary
-    if (rcCommandDeltaAbs) {
-        // movement!
-        if (prevDuplicatePacket[axis] == true) {
-            rxRate = 1.0f / (rxInterval + prevRxInterval);
-            zeroTheAcceleration = 0.0f;
-            // don't add acceleration, empirically seems better on FrSky
-        }
-        setpointSpeed *= rxRate;
-        prevDuplicatePacket[axis] = false;
-    } else {
-        // no movement!
-        if (prevDuplicatePacket[axis] == false) {
-            // first duplicate after movement
-            setpointSpeed = prevSetpointSpeed[axis];
-            if (fabsf(setpoint) < 0.95f * maxRcRate[axis]) {
-                setpointSpeed += prevAcceleration[axis];
+    float setpointSpeed = 0.0f;
+    float setpointAcceleration = 0.0f;
+    float feedforward = 0.0f;
+
+    if (pid->feedforwardInterpolate) {
+        static float prevRxInterval;
+        // for Rx links which do send duplicate data packets when they send telemetry
+        // interpolate setpointSpeed to minimise steps in feedforward
+        if (rcCommandDeltaAbs) {
+            // movement!
+            // if the previous packet was a duplicate, calculate setpointSpeed over the last two intervals
+            if (prevDuplicatePacket[axis] == true) {
```

Comment:
Can be simplified as
```suggestion
            if (prevDuplicatePacket[axis]) {
```

Reply:
OK, thanks